vendor_transport_lead_time/models/product_supplierinfo.py

```diff
@@ -29,10 +32,19 @@ def _compute_delay(self):
 
     def _inverse_delay(self):
         for record in self:
-            diff = record.delay - record.transport_delay
-            if diff < 0:
+            record.supplier_delay = record.delay - record.transport_delay
+
+    @api.constrains("supplier_delay")
+    def _check_delay(self):
+        for seller in self:
+            if seller.supplier_delay < 0:
                 raise ValidationError(
                     _("You can't set a delay inferior to the transport delay.")
                 )
-            else:
-                record.supplier_delay = diff
+
+    @api.model
+    def _setup_fields(self):
+        # "remove" the default lambda on "delay" field (from 'product' module)
+        # to not let Odoo put a value in this field when 'create' is called
+        self._fields["delay"].default = False
+        super()._setup_fields()
```

Comment:
Not sure to understand. Why don't you set the default value directly on the field?

Reply:
Because we have no default value to set on this field (it is computed now), and anyway Odoo will try to set the default value defined in the standard `product`, what we do not want.
I don't define the default to `False` here, I override the generated `lambda` of the field as Odoo is checking this `default` attribute with some `if field.default` here and there in the ORM.